File: frontend/src/services/gpsService.js
import api from './api';

export const gpsService = {
  getLatestLocation: async (vehicleId) => {
    const response = await api.get(`/gps/vehicles/${vehicleId}/latest`);
    return response.data.result;
  },

  getTripHistory: async (tripId) => {
    const response = await api.get(`/gps/trips/${tripId}/history`);
    return response.data.result;
  },

  getVehicleHistory: async (vehicleId) => {
    const response = await api.get(`/gps/vehicles/${vehicleId}/history`);
    return response.data.result;
  },

  simulateMovement: async (vehicleId, tripId) => {
    const response = await api.post(`/gps/vehicles/${vehicleId}/simulate`, null, {
      params: { tripId }
    });
    return response.data.result;
  },

  recordLocation: async (vehicleId, locationData) => {
    const response = await api.post(`/gps/vehicles/${vehicleId}/location`, locationData);
    return response.data.result;
  }
};

File: service/src/main/java/com/FMS/services/GpsService.java
package com.FMS.services;

import com.FMS.dto.VehicleLocationDto;
import com.FMS.dto.request.VehicleLocationRecordRequest;

import java.util.List;

public interface GpsService {
    VehicleLocationDto recordLocation(String vehicleId, VehicleLocationRecordRequest request);
    VehicleLocationDto getLatestLocation(String vehicleId);
    List<VehicleLocationDto> getTripHistory(String tripId);
    List<VehicleLocationDto> getVehicleHistory(String vehicleId);
    VehicleLocationDto simulateMovement(String vehicleId, String tripId);
}
